src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -756,11 +756,11 @@ OverlayImpl::onManifests (
                 protocol::TMManifests o;
                 o.add_list ()->set_stobject (s);
 
-                std::set<Peer::id_t> skip;
-                hashRouter.prepRelay (hash, skip);
-                foreach (send_if_not (
-                    std::make_shared<Message>(o, protocol::mtMANIFESTS),
-                    peer_in_set (skip)));
+                auto const skip = hashRouter.shouldRelay (hash);
+                if(skip)
```

Comment:
Yes, it is. My gut tells me that the old code is wrong. Whenever we get a new manifest, broadcast it to all of our peers. Then each of our peers will broadcast it to all of their peers, which means we'll see it again, several times. Then we broadcast it again... It looks like `applyManifest` will only return `accepted` the first time it's seen, and `stale` thereafter, but I'd rather not rely on that when `HashRouter` is going to be checking anyway. Belt and suspenders.